mangaki/mangaki/utils/mal.py

```diff
@@ -247,6 +310,10 @@ def insert_into_mangaki_database_from_mal(mal_entries: List[MALEntry]):
                 **extra_fields
             )
 
+            if (title and not first_work_matching
+                and title.upper() in list(map(str.upper, titles))):
+                first_work_matching = work
```

Comment:
Do we actually want to add non-matching works? Why isn't this function only creating and returning a matching work, instead of doing that *and* creating a bunch of other works at the same time?